### PR TITLE
XW-3201 | timestamps format update

### PR DESCRIPTION
**Pull request by weronika**
https://wikia-inc.atlassian.net/browse/XW-3201

### CODE REVIEW

File: extensions/wikia/Wall/templates/Wall_message.php [hunk 1]
```diff
@@ -56,6 +56,7 @@
 				<?php if( !empty($isStaff) ): ?>
 					<span class="stafflogo"></span>
 				<?php endif; ?>
+				<span class="subtle edittime" title="<?= $iso_timestamp ?>">• <?= $fmt_timestamp ?></span>
```
Comment (latata) [inline extensions/wikia/Wall/templates/Wall_message.php]: How about moving the bullet outside the `<span>`?

Comment (latata) [reply extensions/wikia/Wall/templates/Wall_message.php]: `edittime` class is not valid here as we show only creation date

File: extensions/GlobalMessages/GlobalMessagesW.i18n.php [hunk 2]
```diff
@@ -208,10 +208,10 @@
 	'wikia-language-top-list' => 'de,en,es,he,fr,it,ja,no,pl,pt,pt-br,zh',
 	'wikia-language-top' => 'Top $1 languages',
 	'wikia-language-all' => 'All languages',
-	'wikia-seconds-ago' => '$1 {{PLURAL:$1|second|seconds}} ago',
-	'wikia-minutes-ago' => '$1 {{PLURAL:$1|minute|minutes}} ago',
-	'wikia-hours-ago' => '$1 {{PLURAL:$1|hour|hours}} ago',
-	'wikia-days-ago' => '$1 {{PLURAL:$1|day|days}} ago',
+	'wikia-seconds-ago' => '$1{{PLURAL:$1|s|s}}',
```
Comment (rybmat) [inline extensions/GlobalMessages/GlobalMessagesW.i18n.php:211]: if the plural and singular version does not differ, do we have to use {{PLURAL}} helper?

File: extensions/wikia/MyHome/templates/feed.tmpl.php [hunk 3]
```diff
@@ -10,7 +10,11 @@
 		<?php } else { ?>
 			<span class="title"><?php print htmlspecialchars($row['title']) ?></span>
 <?php } ?>
-			<cite><?php print FeedRenderer::getActionLabel($row); ?><?php print ActivityFeedRenderer::formatTimestamp($row['timestamp']); ?><?php print FeedRenderer::getDiffLink($row); ?></cite>
+			<?php
+				$timestamp = ActivityFeedRenderer::formatTimestamp( $row['timestamp'] );
+				$timestamp = $timestamp ? ' • ' . $timestamp : '';
+			?>
+			<cite><?php print $timestamp; ?><?php print FeedRenderer::getActionLabel($row); ?><?php print FeedRenderer::getDiffLink($row); ?></cite>
```
Comment (latata) [inline extensions/wikia/MyHome/templates/feed.tmpl.php:17]: Shouldn't we move it to i18n to allow translators make decision about the order of these labels (timestamp / action label)?

File: extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php [hunk 4]
```diff
@@ -38,13 +38,7 @@ public function executeIndex() {
...
 		// format time (for edits older than 30 days - show timestamp)
 		foreach($themeHistory as &$entry) {
-			$diff = time() - strtotime( $entry['timestamp'] );
-
-			if($diff < 30 * 86400) {
-				$entry['timeago'] = wfTimeFormatAgo($entry['timestamp']);
-			} else {
-				$entry['timeago'] = $wgLang->date($entry['timestamp']);
-			}
+			$entry['timeago'] = wfTimeFormatAgo($entry['timestamp']);
```
Comment (rybmat) [inline extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php]: array_map(..)? or at least please avoid modifying array you iterate on

File: extensions/wikia/WikiaNewFiles/WikiaNewFilesGallery.class.php [hunk 5]
```diff
@@ -49,7 +49,7 @@ public function addImages( array $images ) {
 			}
 
 			$caption = wfMessage( 'wikianewfiles-uploadby' )->rawParams( $ul )->params( $ut )->escaped();
-			$caption .= "<br />\n<i>$timeago</i><br />\n";
+			$caption .= " • <i>$timeago</i><br />\n";
```
Comment (latata) [inline extensions/wikia/WikiaNewFiles/WikiaNewFilesGallery.class.php:52]: Don't we want to allow translators to make decision about the order of these labels?

File: includes/wikia/GlobalFunctions.php [hunk 6]
```diff
@@ -927,8 +927,8 @@ function wfTimeFormatAgo( $stamp, $hideCurrentYear = true ) {
 		// Under 24 hrs: to the hour (4 hours ago)
 		$res = wfMsgExt( 'wikia-hours-ago', array( 'parsemag' ), floor( $ago / 3600 ) );
 	}
-	else if ( $ago < 30 * 86400 ) {
-		// Under 30 days: to the day (5 days ago)
+	else if ( $ago < 7 * 86400 ) {
+		// Under 7 days: to the day (5 days ago)
```
Comment (rybmat) [inline includes/wikia/GlobalFunctions.php]: please update the comment to show the actual time format: 5 days ago -> 5d

**Review by rybmat — CHANGES_REQUESTED** [review]

File: extensions/GlobalMessages/GlobalMessagesW.i18n.php (hunk 2 above)
Comment (latata) [reply extensions/GlobalMessages/GlobalMessagesW.i18n.php:211]: we don't know if that won't differ in other languages

**Review by latata — APPROVED** [review]

**Review by rybmat — APPROVED** [review]

File: includes/wikia/GlobalFunctions.php [hunk 7]
```diff
@@ -916,19 +916,19 @@ function wfTimeFormatAgo( $stamp, $hideCurrentYear = true ) {
...
 	}
 	else if ( $ago < 3600 ) {
-		// Under 1 hr: to the minute (3 minutes ago)
+		// Under 1 hr: to the minute (3m)
 		$res = wfMsgExt( 'wikia-minutes-ago', array( 'parsemag' ), floor( $ago / 60 ) );
 	}
 	else if ( $ago < 86400 ) {
-		// Under 24 hrs: to the hour (4 hours ago)
+		// Under 24 hrs: to the hour (4h)
 		$res = wfMsgExt( 'wikia-hours-ago', array( 'parsemag' ), floor( $ago / 3600 ) );
```
Comment (WikiaTech) [inline includes/wikia/GlobalFunctions.php:928]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

File: includes/wikia/GlobalFunctions.php [hunk 8]
```diff
@@ -916,19 +916,19 @@ function wfTimeFormatAgo( $stamp, $hideCurrentYear = true ) {
...
 	}
-	else if ( $ago < 30 * 86400 ) {
-		// Under 30 days: to the day (5 days ago)
+	else if ( $ago < 7 * 86400 ) {
+		// Under 7 days: to the day (5d)
 		$res = wfMsgExt( 'wikia-days-ago', array( 'parsemag' ), floor( $ago / 86400 ) );
```
Comment (WikiaTech) [inline includes/wikia/GlobalFunctions.php:932]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

File: includes/wikia/GlobalFunctions.php [hunk 9]
```diff
@@ -916,19 +916,19 @@ function wfTimeFormatAgo( $stamp, $hideCurrentYear = true ) {
 		// format is date, with a year (July 26, 2008)
 		$res = $wgLang->date( wfTimestamp( TS_MW, $stamp ) );
 	} elseif ( $ago < 60 ) {
-		// Under 1 min: to the second (ex: 30 seconds ago)
+		// Under 1 min: to the second (ex: 30s)
 		$res = wfMsgExt( 'wikia-seconds-ago', array( 'parsemag' ), $ago );
```
Comment (WikiaTech) [inline includes/wikia/GlobalFunctions.php:920]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

File: includes/wikia/GlobalFunctions.php [hunk 10]
```diff
@@ -916,19 +916,19 @@ function wfTimeFormatAgo( $stamp, $hideCurrentYear = true ) {
...
 	}
 	else if ( $ago < 3600 ) {
-		// Under 1 hr: to the minute (3 minutes ago)
+		// Under 1 hr: to the minute (3m)
 		$res = wfMsgExt( 'wikia-minutes-ago', array( 'parsemag' ), floor( $ago / 60 ) );
```
Comment (WikiaTech) [inline includes/wikia/GlobalFunctions.php:924]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 188 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 135 major
* ![MINOR][MINOR] 13 minor
* ![INFO][INFO] 39 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [GlobalFunctions.php#L52](https://github.com/Wikia/app/blob/040f1348b277d131ce5175d8492cff3346125b28/includes/wikia/GlobalFunctions.php#L52): Only the first line of this 2-line block will be executed conditionally. The rest will execute unconditionally. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2681)
1. ![MAJOR][MAJOR] [ThemeDesignerController.class.php#L21](https://github.com/Wikia/app/blob/040f1348b277d131ce5175d8492cff3346125b28/extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php#L21): Remove this unused "$wgLang" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [ThemeDesignerController.class.php#L21](https://github.com/Wikia/app/blob/040f1348b277d131ce5175d8492cff3346125b28/extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php#L21): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ThemeDesignerController.class.php#L100](https://github.com/Wikia/app/blob/040f1348b277d131ce5175d8492cff3346125b28/extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php#L100): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [ThemeDesignerController.class.php#L104](https://github.com/Wikia/app/blob/040f1348b277d131ce5175d8492cff3346125b28/extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php#L104): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [ThemeDesignerController.class.php#L107](https://github.com/Wikia/app/blob/040f1348b277d131ce5175d8492cff3346125b28/extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php#L107): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [ThemeDesignerController.class.php#L110](https://github.com/Wikia/app/blob/040f1348b277d131ce5175d8492cff3346125b28/extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php#L110): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [ThemeDesignerController.class.php#L113](https://github.com/Wikia/app/blob/040f1348b277d131ce5175d8492cff3346125b28/extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php#L113): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [ThemeDesignerController.class.php#L120](https://github.com/Wikia/app/blob/040f1348b277d131ce5175d8492cff3346125b28/extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php#L120): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [ThemeDesignerController.class.php#L126](https://github.com/Wikia/app/blob/040f1348b277d131ce5175d8492cff3346125b28/extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php#L126): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request closed without merging**